pkg/attribute/mutableBag.go

```diff
@@ -56,13 +61,18 @@ var mutableBags = sync.Pool{
 //
 // When you are done using the mutable bag, call the Done method to recycle it.
 func GetMutableBag(parent Bag) *MutableBag {
+	pp, _ := parent.(*MutableBag)
+	if pp != nil {
```

Comment:
There's a race here. The parent could be 'collected' here before the child is attached.